### ISSUE TITLE
Visual Studio 2010 support

### ISSUE THREAD

**Issue by baldurk** [KhronosGroup/glslang#547]
Current master code doesn't build on VS2010. Previously there have been things like range-based for loops that could be converted to iterator loops, syntax sugar like "using x = y;" instead of typedefs, or things like that which are fixable with minor change/impact.

There are some cases like that which could be fixed, and if wanted I can put together a pull request, but it seems now though that there's a fair amount of code using lambdas in the hlsl project which would be a more involved change to remove. Is VS2010 still intended to be supported or has support been dropped?

Note I'm only interested in the actual core glslang lib. E.g. the spv-remapper code has always been heavily C++11 but it just #ifdefs itself out.

**Comment by johnkslang** [comment]
I thought typical lambdas being used worked on VS2010, and have pushed back on anyone adding range-based loops and `{ }` expressions.

What are the examples of new code that doesn't work on VS2010?

**Comment by baldurk** [comment]
Huh, looks like I overreacted - my apologies. I was under the impression that VS2010's lambda support was a poor pre-standardised version but it's more capable than I believed.

Upon digging though it seems like all that's missing (in spite of some really weird misleading error messages) is that there are lambdas relying on being able to guess the return type over multiple statements, which apparently wasn't possible until C++14. Just adding a few liberal `-> bool` etc statements around seems to fix it.

I've now run into some problems in hex_float.h but I think they are fixable with reasonable changes. I'll open a PR with the changes needed when I've got them.
